src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -36506,7 +36556,7 @@ void Main(C c, string? s)
         }
         else
         {
-            s.ToString(); // warn 2
```

Comment:
I think the original behavior was correct. I would not expect the `[NotNullWhen...]` attribute to have any effect on a member that returns `object`, despite that return value later being cast to `bool`.
We have such `bool` checks for some of the nullability attributes (see `ApplyMemberPostConditions`) but it is missing for this scenario. I'll fix that in an upcoming PR shortly.

Reply:
The changes proposed reverted this test. I was not aware this was how it worked, but it make sense.